http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpWsmDao.scala

```diff
@@ -96,38 +106,59 @@ class HttpWsmDao[F[_]](httpClient: Client[F], config: HttpWsmDaoConfig)(implicit
             Uri.Path
               .unsafeFromString(s"/api/workspaces/v1/${workspaceId.value.toString}")
           ),
-        headers = Headers(fakeAuth())
+        headers = Headers(authorization)
       )
     )(onError)
 
-  override def deleteVm(request: DeleteVmRequest)(implicit ev: Ask[F, AppContext]): F[DeleteVmResult] =
-    httpClient.expectOr[DeleteVmResult](
+  override def deleteVm(request: DeleteWsmResourceRequest,
+                        authorization: Authorization)(implicit ev: Ask[F, AppContext]): F[DeleteWsmResourceResult] =
+    deleteHelper(request, authorization, "vm")
+
+  override def deleteDisk(request: DeleteWsmResourceRequest,
+                          authorization: Authorization)(implicit ev: Ask[F, AppContext]): F[DeleteWsmResourceResult] =
+    deleteHelper(request, authorization, "disks")
+
+  override def deleteIp(request: DeleteWsmResourceRequest,
+                        authorization: Authorization)(implicit ev: Ask[F, AppContext]): F[DeleteWsmResourceResult] =
+    deleteHelper(request, authorization, "ip")
+
+  override def deleteNetworks(request: DeleteWsmResourceRequest, authorization: Authorization)(
+    implicit ev: Ask[F, AppContext]
+  ): F[DeleteWsmResourceResult] =
+    deleteHelper(request, authorization, "network")
+
+  override def getCreateVmJobResult(request: GetJobResultRequest, authorization: Authorization)(
+    implicit ev: Ask[F, AppContext]
+  ): F[GetCreateVmJobResult] =
+    httpClient.expectOr[GetCreateVmJobResult](
       Request[F](
-        method = Method.POST,
+        method = Method.GET,
         uri = config.uri
           .withPath(
             Uri.Path
               .unsafeFromString(
-                s"/api/workspaces/v1/${request.workspaceId.value.toString}/resources/controlled/azure/vm/${request.resourceId.value.toString}"
+                s"/api/workspaces/v1/${request.workspaceId.value.toString}/resources/controlled/azure/vm/create-result/${request.jobId.value}"
               )
           ),
-        body = Stream.emits(request.deleteRequest.asJson.noSpaces.getBytes),
-        headers = Headers(fakeAuth())
+        headers = Headers(authorization)
       )
     )(onError)
 
-  override def getCreateVmJobResult(request: GetJobResultRequest)(implicit ev: Ask[F, AppContext]): F[CreateVmResult] =
-    httpClient.expectOr[CreateVmResult](
+  private def deleteHelper(req: DeleteWsmResourceRequest, authorization: Authorization, resource: String)(
```

Comment:
I'd consider using the controlled resource enum [here](https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeControlledResourceComponent.scala#L50) over a string (after possibly moving the enum to a more general location)